lib/review-apps.unit.test.js

```diff
@@ -217,183 +406,111 @@ describe('review-apps', () => {
 				]);
+
 			const build = await getAppBuildWithCommit({ appId, commit });
 			expect(build).toBeFalsy();
+
 		});
 
 		it('returns undefined if there is no builds', async () => {
+
 			const appId = 'app123';
 			const commit = 'ababa';
+
 			nockScope.get('/apps/app123/builds')
 				.reply(200, []);
+
 			const build = await getAppBuildWithCommit({ appId, commit });
 			expect(build).toBeFalsy();
+
 		});
+
 	});
 
-	describe('waitForReviewAppBuild', () => {
-		const { waitForReviewAppBuild } = reviewApps;
+	describe('repeatedCheckForBuildSuccessStatus', () => {
+
+		const { repeatedCheckForBuildSuccessStatus } = reviewApps;
+
+		it('returns without throwing an error if the app build is returned with a status of \'succeeded\'', async () => {
 
-		it('returns appId of succeeded review app build', async () => {
 			const commit = 'a00000000000';
 			const appId = 'app123';
 			const reviewAppBuild = {
-				app: {
-					id: appId
-				},
 				source_blob: {
 					version: commit
 				},
 				status: 'succeeded'
 			};
-			nockScope.get('/apps/app123/builds')
-				.reply(200, [ reviewAppBuild ]);
-
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit
-			})(appId);
 
-			expect(waitedReviewAppBuild).toEqual(appId);
-		});
+			nockScope.get('/apps/app123/builds')
+				.reply(200, [reviewAppBuild]);
 
-		it('waits for succeeded review app build from no review apps', async () => {
-			const commit = 'a00000000000';
-			const appId = 'app123';
-			const succeededReviewAppBuild = {
-				app: {
-					id: appId
-				},
-				source_blob: {
-					version: commit
-				},
-				status: 'succeeded'
-			};
-			nockScope
-				.get('/apps/app123/builds')
-				.reply(200, [])
-				.get('/apps/app123/builds')
-				.reply(200, [ succeededReviewAppBuild ]);
+			try {
+				await repeatedCheckForBuildSuccessStatus({ commit, appId });
+			} catch (error) {
+				expect(true).toBeFalsy();
+			}
 
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit,
-				minTimeout: 0
-			})(appId);
+			expect(true).toBeTruthy();
 
-			expect(waitedReviewAppBuild).toEqual(appId);
 		});
 
-		it('waits for succeeded review app build even with errors', async () => {
+		it('will retry if an error is thrown when attempting to get the app build status', async () => {
+
 			const commit = 'a00000000000';
 			const appId = 'app123';
-			const succeededReviewAppBuild = {
-				app: {
-					id: appId
-				},
-				source_blob: {
-					version: commit
-				},
-				status: 'succeeded'
-			};
-			nockScope
-				.get('/apps/app123/builds')
-				.reply(400, [])
-				.get('/apps/app123/builds')
-				.reply(200, [ succeededReviewAppBuild ]);
 
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit,
-				minTimeout: 0
-			})(appId);
+			nockScope.get('/apps/app123/builds')
+				.times(60)
+				.reply(400, {
+					message: 'Some error occurred'
+				});
+
+			try {
+				await repeatedCheckForBuildSuccessStatus({ commit, appId, minTimeout: 0 });
```

Comment:
Is there a way to check how many times the mock has been called? I know jest has https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber

Reply:
I have used `nock.isDone()`, taken from this blog post: [Ensure All Nock Interceptors Are Used by neverendingqs](https://blog.neverendingqs.com/2017/03/13/ensure-all-nock-interceptors-are-used.html).